Update ModbusMaster.cpp

anklimov opened the pull request:
### Description
Add tolerance to random noise receiving after transmission in half-duplex mode (Invalid Slave 0xE1 error preventing)

Awaiting for 1-s byte of response == SlaveID. All garbage before - discarding

anklimov committed:
Update ModbusMaster.cpp

Add tolerance to random noise receiving after transmission in half-duplex mode (Invalid Slave 0xE1 error preventing)

rnicolas commented:
Hi @anklimov 
I had the same problem, but in my case the noise was always zero, so instead of comparing the readings from serial with the direction in order to know were the frame started, I just discarded all zero values until I read a non 0 value. Do you have random value readings before you read the slave ID?

anklimov commented:
Hi ! 
In my case, I definitely saw non-zero noise values time-by time
Since the fact, correct reply always started with slave ID suggested method looks more reliable
if no appropriate SlaveID received for the response time - Timeout error will occurs 

> On Jan 15, 2018, at 1:16 AM, rnicolas <notifications@github.com> wrote:
> 
> Hi @anklimov <https://github.com/anklimov>
> I had the same problem, but the noise was always zero, so instead of comparing the readings from serial with the direction in order to know were the frame started, I just discarded all zero values until I read a non 0 value.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/4-20ma/ModbusMaster/pull/98#issuecomment-357547179>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AVhOGcbKrkWKOocazokCTR2zzGM9Dk0dks5tKnyngaJpZM4Rdud_>.
>

rnicolas commented:
@anklimov 
Ok! Thanks! I'll update my code to be like yours because it's more generic.

anklimov committed:
Rename crc16.h to crc16_.h

anklimov committed:
Resolve Bridge.cpp compilation issue

Resolve "Bridge.cpp:121:33: error: '_crc_ccitt_update' was not declared in this scope"

anklimov committed:
MODBUS_DEBUG compilation option added

anklimov committed:
debugSerialPort definition

anklimov committed:
raw data transactions

anklimov committed:
Raw transaction - adding device address in 1-st byte

anklimov committed:
re-check serial input after idle handler to prevent timeouts

anklimov committed:
methods to replace response buffer
